### ISSUE TITLE
Auto provide facebook app out of the box

### ISSUE THREAD

**Issue by m-vo** [m-vo/contao-facebook-import#16]
Memo to myself:

Reiterate if the bundle could auto provide a facebook app. This would prevent things like #15, #3, ... and ease the setup procedure in general. 

Todo: 
* [x] check app quota vs page quota
* [ ] find out the implications of misuse
* [ ] find out how to obtain an access token

-----
from https://developers.facebook.com/docs/graph-api/overview/rate-limiting/ about *Application-Level Rate Limiting*:

>These limits apply to calls made using any access token other than a Page access token.
>
>Rate limting is in real time at the app level over a given time range. The total number of calls your app can make per hour is 200 times the number of Users. This is not a per User limit. Any individual User can make more than 200 calls per hour, as long as the total for all Users does not exceed the app maximum. For example, if your app has 100 Users, the app can make 20,000 calls per hour. However, your top ten most engaged Users could make 19,000 of those calls.
>
> [&hellip;]

**Comment by fritzmg** [comment]
Would that work at all? I mean for each Facebook app you have to configure individual app domains and valid login callback urls.

**Comment by m-vo** [comment]
I'm not sure if these things are only needed for login via FB?

**Comment by m-vo** [comment]
This WordPress plugin e.g. promises that there is no app required https://de.wordpress.org/plugins/custom-facebook-feed/ 🤔

**Comment by fritzmg** [comment]
Yeah, it looks like they have set up an app which has the following app domain and callback URL configured: https://api.smashballoon.com/facebook-login.php
When hitting the login button in the WordPress backend, the user is logged in there and then that application provides a response with a user access token, presumably.

**Comment by m-vo** [comment]
How would you implement this without a server? Shipping such a functionality would mean publishing the app's secret, right?

The per-app quota (see updated issue) implies a 200 calls / hour / user (access token) limit on top of the current one. This is no problem for the bundle to cater for but it could be exploited by a user as the limit really isn't enforced on a per user basis but rather on a per app basis (200 * user count).

**Comment by fritzmg** [comment]
> How would you implement this without a server?

You can't :)

> Shipping such a functionality would mean publishing the app's secret, right?

Hm, I guess so, yeah.

**Comment by m-vo** [comment]
Sooo, anyone willing to provide a server? :smile_cat:
